### PR TITLE
[nextest] Add retries and don fail fast in CI

### PR DESCRIPTION
**Pull request by perryjrandall**
Its an unfortunate reality of life that unittests can be flaky, especially when
they're creating databases or interacting with files or both!

To deal with this reality lets not fail PR / main when we have a flaky test,
however in tandem we need to monitor the flaky test in main and work to remove
the flakiness.

Test Plan: 

https://github.com/aptos-labs/aptos-core/runs/7983894955?check_suite_focus=true

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3360)

### CODE REVIEW

**Review by rustielin — APPROVED** [review]
LGTM

File: .github/workflows/lint-test.yaml [hunk 1]
```diff
@@ -147,7 +147,7 @@ jobs:
       - uses: taiki-e/install-action@v1.5.6
         with:
           tool: nextest
-      - run: cargo nextest run --profile ci --workspace --exclude smoke-test --exclude backup-cli --exclude testcases
+      - run: cargo nextest run --profile ci --workspace --exclude smoke-test --exclude backup-cli --exclude testcases --retries 3 --no-fail-fast
```
Comment (zekun000) [inline .github/workflows/lint-test.yaml:150]: i thought we already have retries to 3? https://github.com/aptos-labs/aptos-core/blob/6b4c39d61266e224c84ae1c02d2d967c6ca4ecf6/.config/nextest.toml#L13

it's in nextest.toml

Comment (perryjrandall) [reply .github/workflows/lint-test.yaml:150]: That is for smoke test, not unittest

Comment (perryjrandall) [reply .github/workflows/lint-test.yaml:150]: Also actually smoke test doesnt even use this, this is for smoke coverage

Smoke test also use retries flag on command line in workflow as well :P

**Pull request merged**